vico/multiplatform/src/commonMain/kotlin/com/patrykandpatrick/vico/multiplatform/cartesian/axis/HorizontalAxis.kt

```diff
@@ -113,7 +117,36 @@ protected constructor(
     title,
   )
 
-  override fun drawUnderLayers(context: CartesianDrawingContext) {
+  override fun updateAxisDimensions(
+    context: CartesianDrawingContext,
+    axisDimensions: MutableAxisDimensions,
+  ) {
+    with(context) {
+      val lineExtensionLength =
+        if (itemPlacer.getShiftExtremeLines(context)) {
+          tickThickness
+        } else {
+          tickThickness.half
+        }
+      val top =
+        if (position == Axis.Position.Horizontal.Top) {
+          bounds.bottom
+        } else {
+          bounds.top
+        }
+      axisDimensions.lineBounds.set(
+        layerBounds.left - lineExtensionLength,
+        top,
+        layerBounds.right + lineExtensionLength,
+        top + lineThickness,
+      )
```

Comment:
This logic seems incorrect for top axes. We report `bounds.bottom` for the top of the axis line, and `bounds.bottom + lineThickness` for the bottom; the correct values are `bounds.bottom - lineThickness` and `bounds.bottom`, respectively.

```suggestion
      axisDimensions.lineBounds.set(
        layerBounds.left - lineExtensionLength,
        if (position == Axis.Position.Horizontal.Top) bounds.bottom - lineThickness else bounds.top,
        layerBounds.right + lineExtensionLength,
        if (position == Axis.Position.Horizontal.Top) bounds.bottom else bounds.top + lineThickness,
      )
```